pkg/promtail/target.go

```diff
@@ -252,11 +271,26 @@ func (t *tailer) run() {
 			if err := t.handler.Handle(model.LabelSet{}, line.Time, line.Text); err != nil {
 				level.Error(t.logger).Log("msg", "error handling line", "error", err)
 			}
+		case <-t.quit:
+			return
 		}
 	}
 }
 
+func (t *tailer) markPosition() error {
+	pos, err := t.tail.Tell()
+	if err != nil {
+		level.Error(t.logger).Log("msg", "error getting tail position", "error", err)
+		return err
+	}
+	level.Debug(t.logger).Log("path", t.path, "current_position", pos)
+	t.positions.Put(t.path, pos)
+	return nil
+}
+
 func (t *tailer) stop() error {
+	close(t.quit)
+	<-t.quitComplete
 	return t.tail.Stop()
```

Comment:
Is the extra `quit` channel necessary, or can we use `t.tail.Stop()`?

Reply:
this channels is necessary to update the current position with any changes since the last `positionSyncPeriod`, otherwise they are lost